lib/chef/resource/macos_userdefaults.rb

```diff
@@ -78,7 +79,7 @@ class MacosUserDefaults < Chef::Resource
         description: "The preference key.",
         required: true
 
-      property :host, [String, Symbol],
+      property :host, [String, Symbol, FFI::Pointer],
```

Comment:
So the problem with this is that this shows up in Documentation and you're basically saying its legal an expected for the end chef user to pass in an FFI::Pointer object which is really an internal representation and not something we expect users to supply.

That suggests that the coercion here is probably the wrong place, it should get moved somewhere into the provider/action and the String/Symbol should be the representation on the new_resource (and on the current_resource/after_resource as its loaded in load_current_value).

Reply:
Removed the coerce property and now the resource just calls the methods whenever it needs to transform. moving in the action would duplicate the imports and writing a provider seemed like an overkill. Please let me know if https://github.com/chef/chef/pull/12234/commits/03b421f4fec907109d19c2ac00a9b445e65c7849 looks fine.